server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillDataSet.java

```diff
@@ -174,13 +177,18 @@ public RowRecord nextWithoutConstraint() throws IOException {
     try {
       slideCache(record.getTimestamp());
     } catch (QueryProcessException e) {
-      logger.warn("group by fill has an exception while sliding: ", e);
+      logger.error("group by fill has an exception while sliding: ", e);
     }
 
     return record;
   }
 
   private void fillRecord(int index, RowRecord record) throws IOException {
+    if (unsupportedFillMethod[index]) {
```

Comment:
It's better to add test cases to cover unsatisfied data types with `GROUP BY FILL`.

Reply:
I've added tests in last commit